### PR TITLE
Deprecate imread() reading from URLs

### LINKED ISSUE #18648: Drop support for directly imread()ing urls.
### Describe the issue

**Summary**

Support for directly imread()ing url strings caused Matplotlib to add a dependency on a package which we should not have no business of having a dependency on (`certifi`, see https://github.com/matplotlib/matplotlib/pull/18636#issuecomment-703126749).  url strings can't be directly `open`'ed either, so if we instead documented the (easy) solution of calling `urllib.request.urlopen` on the URL (+/- certifi) it'd be much more educative for users (now they know how to urlopen anything, not just images) (note that one of the main reasons for originally adding support to directly opening url strings was that the name of `urllib.request.urlopen` changed between Py2 and Py3, https://github.com/matplotlib/matplotlib/pull/4256#issuecomment-84369830, but that's obviously not relevant anymore).  Finally if we really want that functionality it would make much more sense to have it in Pillow anyways (given that imread is just a thin wrapper around PIL.Image.open at that point).

### Proposed fix

Deprecate and then drop support for `imread("https://...")`.

### PR DESCRIPTION
**Pull request by anntzer**
## PR Summary

Replaces https://github.com/matplotlib/matplotlib/pull/18649, with the suggestion to directly using the Pillow API (https://github.com/matplotlib/matplotlib/pull/18649#pullrequestreview-503145250).  Closes #18648.

## PR Checklist

- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [ ] New features are documented, with examples if plot related.
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

File: lib/matplotlib/tests/test_image.py [hunk 1]
```diff
@@ -1128,7 +1129,8 @@ def test_exact_vmin():
 @pytest.mark.network
 @pytest.mark.flaky
 def test_https_imread_smoketest():
-    v = mimage.imread('https://matplotlib.org/1.5.0/_static/logo2.png')
+    with _api.suppress_matplotlib_deprecation_warning():
+        v = mimage.imread('https://matplotlib.org/1.5.0/_static/logo2.png')
```
Comment (jklymak) [inline lib/matplotlib/tests/test_image.py:1133]: Shouldn't you properly fix this?

**Comment by jklymak** [comment]
I pushed a point release, but feel free to push back if this is really urgent.

File: lib/matplotlib/tests/test_image.py (hunk 1 above)
Comment (anntzer) [reply lib/matplotlib/tests/test_image.py:1133]: The point is that after the removal, reading this just won't involve matplotlib at all (it'll be `PIL.Image.open(urllib.request.urlopen(...))`, so the whole test will go away.

Comment (jklymak) [reply lib/matplotlib/tests/test_image.py:1133]: oops sorry - I missed what the test was for....

**Comment by anntzer** [comment]
I don't think we should introduce new deprecation in bugfix releases, so this should either be 3.4.0, or, if it can't make it, 3.5.0.

**Review by jklymak — APPROVED** [review]
Seems fine.  Feel free to re0milestone if there is a second review

**Review by timhoffm — APPROVED** [review]

**Pull request merged**